File: hw1/HW1Tests/CalculatorTests.cs
using System;
using hw1;
using Xunit;

namespace HW1Tests
{
    public class CalculatorTests
    {
        [Theory]
        [InlineData(Operations.Plus, 6, 9, 15)]
        [InlineData(Operations.Plus, -2, 9, 7)]
        [InlineData(Operations.Plus, 17, -7, 10)]
        [InlineData(Operations.Plus, -3, -5, -8)]
        public void Calculate_Plus_Return_Correct_Result(Operations operation,
            int val1, int val2, int excepted)
        {
            var result = Calculator.Calculate(operation, val1, val2);
            Assert.Equal(excepted, result);
        }
        
        [Theory]
        [InlineData(Operations.Minus, 5, 7, -2)]
        [InlineData(Operations.Minus, 10, -7, 17)]
        [InlineData(Operations.Minus, -7, 7, -14)]
        [InlineData(Operations.Minus, -25, -36, 11)]
        public void Calculate_Minus_Return_Correct_Result(Operations operation,
            int val1, int val2, int excepted)
        {
            var result = Calculator.Calculate(operation, val1, val2);
            Assert.Equal(excepted, result);
        }
        
        [Theory]
        [InlineData(Operations.Multiply, 4, 3, 12)]
        [InlineData(Operations.Multiply, 6, -3, -18)]
        [InlineData(Operations.Multiply, -5, 6, -30)]
        [InlineData(Operations.Multiply, -7, -7, 49)]
        public void Calculate_Multiply_Return_Correct_Result(Operations operation,
            int val1, int val2, int excepted)
        {
            var result = Calculator.Calculate(operation, val1, val2);
            Assert.Equal(excepted, result);
        }
        
        [Theory]
        [InlineData(Operations.Divide, 6, 2, 3)]
        [InlineData(Operations.Divide, 7, -1, -7)]
        [InlineData(Operations.Divide, -10, 5, -2)]
        [InlineData(Operations.Divide, -24, -6, 4)]
        public void Calculate_Divide_Return_Correct_Result(Operations operation,
            int val1, int val2, int excepted)
        {
            var result = Calculator.Calculate(operation, val1, val2);
            Assert.Equal(excepted, result);
        }
    }
}

File: HW6/HW6TestFS/CalculatorTests.fs
module HW6Tests.CalculatorTests

open Xunit
open HW5
open HW5.ResultBuilder


[<Theory>]
[<InlineData(6.1, 8.9, 15)>]
[<InlineData(17.3, -7.3, 10)>]
[<InlineData(-2.6, 9.6, 7)>]
[<InlineData(-3.37, -4.63, -8)>]
let Calculate_Plus_WillReturnCorrectResult (val1, val2, expected) =
     let result = Calculator.calculate (val1, Operations.Plus, val2)
     Assert.Equal(expected, result)
    
    
[<Theory>]
[<InlineData(5, 7, -2)>]
[<InlineData(10, -7, 17)>]
[<InlineData(-7, 7, -14)>]
[<InlineData(-25, -36, 11)>]
let Calculate_Minus_WillReturnCorrectResult (val1, val2, expected) = 
    let result = Calculator.calculate (val1, Operations.Minus, val2)
    Assert.Equal(expected, result)
    
[<Theory>]
[<InlineData(4, 3, 12)>]
[<InlineData(6, -3, -18)>]
[<InlineData(-5, 6, -30)>]
[<InlineData(-7, -7, 49)>]
let Calculate_Multiply_WillReturnCorrectResult (val1, val2, expected) = 
    let result = Calculator.calculate (val1, Operations.Multiply, val2)
    Assert.Equal(expected, result)

[<Theory>]
[<InlineData(6, 2, 3)>]
[<InlineData(7, -1, -7)>]
[<InlineData(-10, 5, -2)>]
[<InlineData(-24, -6, 4)>]
[<InlineData(7, 2, 3.5)>]
let Calculate_Divide_WillReturnCorrectResult (val1,  val2, expected) = 
    let result = Calculator.calculate (val1, Operations.Divide, val2)
    Assert.Equal(expected, result)

